build/docker-compose.yaml

```diff
@@ -53,20 +55,28 @@ services:
 
   snapshot-cmake: &snapshot-cmake
     <<: *build-setup
-    command: bash -c 'if [ -f CMakeLists.txt ]; then mkdir -p work && cd work && cmake .. && make -j "$${MAKEJOBS}"; fi'
+    command: bash -c 'if [ -f CMakeLists.txt ]; then mkdir -p "$${BUILD_DIR}" && cd "$${BUILD_DIR}" && cmake .. && make -j "$${MAKEJOBS}"; fi'
 
   prb-cmake:
     <<: *snapshot-cmake
 
 
   snapshot-ctest: &snapshot-ctest
     <<: *build-setup
-    command: bash -c 'if [ -f CMakeLists.txt ]; then mkdir -p work && cd work && cmake .. && make -j "$${MAKEJOBS}" && ctest; fi'
+    command: bash -c 'if [ -f CMakeLists.txt ]; then mkdir -p "$${BUILD_DIR}" && cd "$${BUILD_DIR}" && cmake .. && make -j "$${MAKEJOBS}" && ctest -L fast -j "$${MAKEJOBS}" --output-on-failure; fi'
 
   prb-ctest:
     <<: *snapshot-ctest
 
 
+  snapshot-correctness: &snapshot-correctness
```

Comment:
This makes it sound like ctest isn't correctness.  `snapshot-ctest-fast` and `snapshot-ctest-full` ?

Reply:
IMO: ctest-full is actually the current implementation of correctness just like ctest-fast is the current implementation of demonstrating ctest. Both of theses implementations may change in the future or in some cases even go away but the name will stay the same.